tide_station_locator.module

```diff
@@ -52,6 +53,50 @@ function tide_station_locator_migrate_stations($options) {
     // Something went wrong somewhere.
     watchdog_exception('tide_station_locator', $e);
   }
+
+  // Set the correct moderation of stations based on dates.
+  _tide_station_locator_content_moderator();
+}
+
+/**
+ * Helper method to set the correct moderation based on dates.
+ */
+function _tide_station_locator_content_moderator() {
+  $nids = \Drupal::entityQuery('node')
+    ->condition('type', 'station')
+    ->accessCheck(FALSE)
+    ->execute();
+
+  $stations = Node::loadMultiple($nids);
+
+  foreach ($stations as $station) {
+    $from = $station->get('field_validity_from')->getValue()[0]['value'] ?? null;
+    $to = $station->get('field_validity_to')->getValue()[0]['value'] ?? null;
+    $current_time = time();
+
+    if ($from && $to) {
+      $from_time = strtotime($from);
+      $to_time = strtotime($to);
+
+      if ($from_time < $current_time && $to_time > $current_time) {
+        $moderation_state = 'published';
+      }
+      elseif ($from_time > $current_time) {
+        $moderation_state = 'draft';
+      }
+      elseif ($to_time < $current_time) {
+        $moderation_state = 'archived';
+      }
+
+      if (isset($moderation_state)) {
```

Comment:
@g2410g Can you add a check here? If the current state is same as the new stata - we do not need to process this station.
This will help in optimising the node save.

Reply:
@gargsuchi Updated the PR, please review.